### ISSUE TITLE
Segmentation fault after first epoch

### ISSUE THREAD

**Issue by Leandree** [apple/tensorflow_macos#127]
tried CPU / ANY and still end up with a segmentation fault at the end of the first epoch. 
When I tried with GPU, the segmentation fault appeared at the start of the first epoch.
This python script work on my other tensorflow environment.

```
inp = Input(shape = (672,37))
l1 = Conv1D(128, 4, activation="relu", padding="same")(inp)
l2 = Dropout(0.2)(l1)
l3 = Conv1D(256, 4, activation="relu", padding="same")(l2)
l4 = Dropout(0.2)(l3)
l5 = Conv1D(512, 4, activation="relu", padding="same")(l4)
l6 = Dropout(0.2)(l5)
l7 = Bidirectional(LSTM(20, return_sequences=True))(l6)
out = Dense(8, activation="sigmoid")(l7)

model = Model(inputs = inp, outputs = out)

model.compile(optimizer = 'adam', loss = 'mean_squared_error')

model.summary()


model.fit(x_train, y_train, epochs=10, validation_data=(x_test, y_test))
```

<img width="713" alt="Capture d’écran 2021-01-11 à 17 31 16" src="https://user-images.githubusercontent.com/32459191/104210199-e14dc380-5432-11eb-9044-6e1f3151c39c.png">

**Comment by anna-tikhonova** [comment]
Thank you very much for reporting this. We will investigate and report back.

**Comment by atw1020** [comment]
I am experiencing a similar issue and can provide a dataset to replicate and environment installation specifics

**Installation:** I followed [This guide](https://towardsdatascience.com/tensorflow-2-4-on-apple-silicon-m1-installation-under-conda-environment-ba6de962b3b8) to create a conda environment which the program is run through.

**[Project on which this can be reproduced](https://github.com/atw1020/tensorflowBug)**

/Users/arthur/miniforge3/envs/tf24/bin/python /Users/arthur/PycharmProjects/tensorflowBug/bug.py
Found 3285 files belonging to 5 classes.
Epoch 1/40
2021-01-24 09:41:31.206991: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)
2021-01-24 09:41:31.208070: W tensorflow/core/platform/profile_utils/cpu_utils.cc:126] Failed to get CPU frequency: 0 Hz
102/103 [============================>.] - ETA: 0s - loss: 17.6890 - accuracy: 0.2257Fatal Python error: Fatal Python error: Segmentation fault
Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

**Comment by atw1020** [comment]
This papers to be patched in the February 2 Patch, Try updating

**Comment by atw1020** [comment]
I just ran this on the latest patch and can conform that the February 1 Patch did not fix this issue. However I also found a workaround


This issue Appears to be caused by batch size, where larger batch sizes lead to this issue. Using a smaller batch size or Stochastic Gradient Descent will resolve this issue. It also appears that the size at which batches become "too big" differs from application to application, so I suggest decreasing your batch size by powers of two until you find something that does not cause a segmentation fault.

**NOTE**: Adjusting batch size will also likely mean that you need to re-select some hyper parameters such as dropout rate and learning rate.

This also suggests the possibility that this error can be replicated on any network by increasing the batch size.

**Comment by nsleesanghyub** [comment]
i got a same issue on m1 mac at pycharm: Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

'decrease batch size' solve this problem. thank atw1020

**Comment by atw1020** [comment]
The entire issue is resolved in 0.3-alpha1 so make sure you update
